runtime/jcl/common/system.c

```diff
@@ -144,6 +142,12 @@ Java_java_lang_System_getSysPropBeforePropertiesInitialized(JNIEnv *env, jclass
 			sysPropValue = "UTF-8";
 #else /* JAVA_SPEC_VERSION >= 18 */
 			sysPropValue = getPlatformFileEncoding(env, property, sizeof(property), sysPropID);
+#endif /* JAVA_SPEC_VERSION >= 18 */
+		} else {
+#if JAVA_SPEC_VERSION >= 18
```

Comment:
There is a check for this above, this block of code could just move up to it and then there won't be an empty else case when < 18.

Reply:
fixed.